### PR TITLE
Docs(web-react, web-twig): Add Button demo #DS-892

### PR DESCRIPTION
**Pull request by pavelklibani**
## Description

Button demo in web-react is now same as demo in web

### Issue reference

[Demo: Button má ve všech technologiích stejná dema](https://jira.lmc.cz/browse/DS-892)

### CODE REVIEW

File: packages/web-react/src/components/Button/demo/createButtonDemo.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { Button } from '../Button';
+import { Icon } from '../../Icon';
+import { ButtonBaseProps } from '../../../types';
+
+const createButtonDemo = (labels: [string, string], props?: ButtonBaseProps) => {
+  const sizes = ['small', 'medium', 'large'];
```
Comment (literat) [inline packages/web-react/src/components/Button/demo/createButtonDemo.tsx:7]: We have dictionariers/constants for this, so please use them ;-)

File: packages/web-react/src/components/Button/demo/createButtonDemo.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { Button } from '../Button';
+import { Icon } from '../../Icon';
+import { ButtonBaseProps } from '../../../types';
+
+const createButtonDemo = (labels: [string, string], props?: ButtonBaseProps) => {
```
Comment (literat) [inline packages/web-react/src/components/Button/demo/createButtonDemo.tsx:6]: What about to make a hook from this? `useDemoButtons`? Not quite sure about this but it seems fine to me :-) What do you think?

Comment (pavelklibani) [reply packages/web-react/src/components/Button/demo/createButtonDemo.tsx:6]: I always thought that returning JSX from hooks is an antipattern. But I can rework it.

File: packages/web-react/src/components/Button/demo/createButtonDemo.tsx (hunk 1 above)
Comment (pavelklibani) [reply packages/web-react/src/components/Button/demo/createButtonDemo.tsx:7]: Good idea

File: packages/web-react/src/components/Button/demo/createButtonDemo.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  const sizes = ['small', 'medium', 'large'];
+  const colors = ['primary', 'secondary', 'tertiary', 'inverted', 'success', 'informative', 'warning', 'danger'];
+  const [label, label2] = labels;
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/components/Button/demo/createButtonDemo.tsx:9]: we have two labels (`Button/Disabled {color}` and `Menu`) in demo, but i can reduce it to one... the "Menu" is not changing across demo

image

[the remark this answers is not in the document]
Comment (literat) [reply packages/web-react/src/components/Button/demo/createButtonDemo.tsx:9]: Ok, then hardcode `Menu`, please.

**Comment by literat** [comment]
I see a little glitch between https://spirit-design-system-demo.netlify.app/src/scss/components/button/ and https://deploy-preview-1024--spirit-design-system-react.netlify.app/src/components/button/. But I do not know why. Maybe @adamkudrna  or @crishpeen will know?

File: packages/web-react/src/components/Button/demo/createButtonDemo.tsx (hunk 2 above)
Comment (literat) [reply packages/web-react/src/components/Button/demo/createButtonDemo.tsx:6]: No, you are right. In fact, this is a "factory" function, see https://github.com/themithy/react-design-patterns/blob/master/doc/factory-pattern.md. So I am thinking now just about the naming if we should use `factory` in the name or `render` or something else. 🤔

Comment (pavelklibani) [reply packages/web-react/src/components/Button/demo/createButtonDemo.tsx]: Created new component `<ButtonDemoFactory />`, its better than the function before, thank you.

**Comment by crishpeen** [comment]
> I see a little glitch between https://spirit-design-system-demo.netlify.app/src/scss/components/button/ and https://deploy-preview-1024--spirit-design-system-react.netlify.app/src/components/button/. But I do not know why. Maybe @adamkudrna or @crishpeen will know?

https://github.com/facebook/react/issues/1643

File: packages/web-react/src/components/Button/demo/ButtonDemoFactory.tsx
Comment (literat) [inline packages/web-react/src/components/Button/demo/ButtonDemoFactory.tsx:1]: 👍

**Review by literat — APPROVED** [review]

File: packages/web-twig/src/Resources/components/Button/stories/ButtonLoading.twig [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<section class="docs-Section">
+  <h2 class="docs-Heading">Disabled</h2>
+
+  <Grid cols="1" tablet="2" desktop="3">
+    {% for size in sizes %}
+    <div>
+      <h3 class="docs-Heading">Size {{size}}</h3>
```
Comment (crishpeen) [inline packages/web-twig/src/Resources/components/Button/stories/ButtonLoading.twig:7]:
```suggestion
      <h3 class="docs-Heading">Size {{ size }}</h3>
```

File: packages/web-react/src/components/Button/demo/ButtonDemoFactory.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { Button } from '../Button';
+import { Icon } from '../../Icon';
+import { ButtonBaseProps } from '../../../types';
+import { Sizes, ActionColors, EmotionColors } from '../../../constants/dictionaries';
```
[the remark this answers is not in the document]
Comment (pavelklibani) [reply packages/web-react/src/components/Button/demo/ButtonDemoFactory.tsx:5]: Sorted! 👍 Sorry and thanks for pointing these out; it's helping me remember those things! 😁

File: packages/web-twig/src/Resources/components/Button/stories/ButtonLoading.twig (hunk 4 above)
Comment (pavelklibani) [reply packages/web-twig/src/Resources/components/Button/stories/ButtonLoading.twig:7]: fixed

File: packages/web-twig/src/Resources/components/Button/stories/ButtonLoading.twig [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      {% for color in colors %}
+      <div class="{% if color is same as('inverted') %}docs-Box {% endif %}my-600">
+        <div>
+          <Button color={color} size={size} isLoading>Loading {{ color }}</Button>
+          <Button color={color} size={size} isLoading>
+            <Icon name="hamburger" UNSAFE_className="mr-400" />
+            Menu
+          </Button>
+          <Button color={color} size={size} isSquare isLoading>
+            <span class="accessibility-hidden">Menu</span>
```
Comment (crishpeen) [inline packages/web-twig/src/Resources/components/Button/stories/ButtonLoading.twig:17]: VisuallyHidden component? :)

File: packages/web-twig/src/Resources/components/Button/stories/ButtonLoading.twig [hunk 7]
```diff
@@ -0,0 +1,26 @@
+<section class="docs-Section">
+  <h2 class="docs-Heading">Disabled</h2>
```
Comment (crishpeen) [inline packages/web-twig/src/Resources/components/Button/stories/ButtonLoading.twig:2]: The file is called Loading, but the title is Disabled, probably the wrong title.

**Review by crishpeen — APPROVED** [review]

File: packages/web-twig/src/Resources/components/Button/stories/ButtonLoading.twig (hunk 7 above)
Comment (pavelklibani) [reply packages/web-twig/src/Resources/components/Button/stories/ButtonLoading.twig:2]: fixed, also moved all titles from demo files to .stories.twig file

**Pull request merged**